### DISCUSSION TITLE
Trigger error and loading for creating custom devtools

### DISCUSSION THREAD

**Discussion by nicoandreoli2000** [TanStack/query#7692] [Ideas]
Hello Tanstack team :D

I'm trying to develop some custom devtools to show loading and error states to UX team.. Unfortunately, I believe there's no actual way of doing this right now. So what I suggest is that tanstack query devtools exposes some functions or hooks that can be used to manipulate things like error and loading states.

Let me know your thoughts!

**Comment by nicoandreoli2000** [comment]
Btw it will be great if this actually can be done but idk how, if that's the case please let me know. I tried with `setQueryDefaultOptions` but doesn't work.

**Comment by TkDodo** [comment]
> I'm trying to develop some custom devtools to show loading and error states to UX team.

our devtools already allow to put your query in pending or error state
